agents/README.md

```diff
@@ -0,0 +1,223 @@
+
+We can create that namespace with the following command:
+
+```bash
+NAMESPACE="rl"
+kubectl create namespace ${NAMESPACE}
+```
+
+### Running the demo on JupyterHub
+
+In order to run this demo on a Kubeflow JupyterHub deployment you'll need the registry address of a demo container that is accessible to that Kubeflow deployment. This can be accomplished using the public container image for this demo, `quay.io/cwbeitel/agents-demo:0.3`, or by building the demo container yourself.
+
+Please refer to [these docs](https://github.com/kubeflow/kubeflow/blob/master/user_guide.md) for accessing your JupyterHub deployment. In short you should be able to access JupyterHub in your browser at http://127.0.0.1:8000 after running:
+
+```bash
+PODNAME=`kubectl get pods --namespace=kubeflow --selector="app=tf-hub" --output=template --template="{{with index .items 0}}{{.metadata.name}}{{end}}"`
+kubectl port-forward --namespace=kubeflow $PODNAME 8000:8000
+```
+
+At which point you can use any username and password to sign in. (Relatedly, the default service type for JupyterHub on Kubeflow ClusterIP which means that by default it is not exposed via a public IP.)
+
+##### Building your own development container
+
+We might want to include various additional dependencies in our notebook container to aid in development of our model. In that case we might be interested in modifying the demo Dockerfile and re-building the container.
+
+The demo container can be built using the standard workflow:
+
+```bash
+TAG=quay.io/someuser/agents-demo:version
+docker build -t ${TAG} .
+docker push ${TAG}
+```
+
+##### Launching
+
+To run the examples on a Kubeflow JupyterHub deployment simply provide the registry address of your demonstration container, e.g. `quay.io/cwbeitel/agents-demo:0.3`, in the image field of the spawner options dialog.
+
+Here's approximately what that looks like for me:
+
+![](demo/jhub-spawn.png)
+
+Once you've accessed your Jupyter server you will be ready to upload service account credentials that will allow you to perform various operations on Kubernetes and other cloud services. In the IAM section of your GCP account these can be configured as follows:
+
+![](demo/sa-create.png)
+
+can upload service account credentials from your local machine to your JupyterHub instance via Jupyter's file upload dialog (the upload button in the upper right of your Jupyter home screen).
+
+
+
+This will by default place the uploaded credentials at /home/joyvan/[your-key].json which you'll use in the initial setup steps of the demonstration.
+
+Please note that if the spawn of your notebook hangs with "Your server is starting up..." without ever spawning the notebook you can troubleshoot by describing the state of the underlying pod. We can get the name of the underlying pod from the list of Kubeflow pods based on the username used to log into JupyterHub:
```

Comment:
This troubleshooting info doesn't seem specific to this example.
Can we move it to the troubleshooting section of the user_guide?

Reply:
Yes